pkg/config/structure/unmarshal.go

```diff
@@ -49,7 +49,12 @@ func UnmarshalKey(cfg model.Reader, key string, target interface{}, opts ...Unma
 	for _, o := range opts {
 		o(fs)
 	}
-	source, err := newNode(reflect.ValueOf(cfg.Get(key)))
+	rawval := cfg.Get(key)
+	// Don't create a reflect.Value out of nil, just return immediately
+	if rawval == nil {
```

Comment:
`setupAutoDiscovery` in `cmd/agent/common/autodiscovery.go` expects a nil config setting to return no error. Add this fix gets the `new-e2e-aml` test to pass.